### PR TITLE
Cruciform forge

### PR DESCRIPTION
**Pull request by DarkLevelSS13**
## About The Pull Request
This is a unique structure that cannot be rebuilt if destroyed. While it can’t be dismantled, it can be destroyed with sufficient force (explosives, most likely).

When fed biomatter, gold, and plasteel, the forge will spit out blank cruciforms to be used in Obelisk creation and converting more followers.

As with all NT machinery, it will be controlled via litany - like the biogenerator.
image
image

## Changelog
:cl: DarkLevel - code, CeUvi - sprites.
add: Cruciform Forge
/:cl:

### CODE REVIEW

**Review by genessee596 — APPROVED** [review]

File: code/modules/core_implant/cruciform/rituals/machinery.dm [hunk 1]
```diff
@@ -33,7 +33,39 @@
...
+	if(forge.working)
+		fail("[forge] is already working!", user, C)
+		return FALSE
+
+	if(forge.stat & NOPOWER)
+		fail("[forge] is off.", user, C)
+		return FALSE
+
+	for(var/_material in forge.needed_material)
+		if(!(_material in  forge.stored_material))
```
Comment (Evankhell561) [inline code/modules/core_implant/cruciform/rituals/machinery.dm:59]:
```suggestion
		if(!(_material in forge.stored_material))
```

**Review by Evankhell561 — CHANGES_REQUESTED** [review]

**Review by Evankhell561 — CHANGES_REQUESTED** [review]

File: code/modules/core_implant/cruciform/machinery/cruciformforge.dm [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+	density = TRUE
+	anchored = TRUE
+	layer = 2.8
+
+	var/power_cost = 250
+
+	var/working = FALSE
+	var/start_working = null
+	var/work_time = 30 SECONDS
+	var/paused = FALSE
```
Comment (Evankhell561) [inline code/modules/core_implant/cruciform/machinery/cruciformforge.dm:20]: why?

**Review by Evankhell561 — CHANGES_REQUESTED** [review]

File: code/modules/core_implant/cruciform/machinery/cruciformforge.dm [hunk 3]
```diff
@@ -0,0 +1,192 @@
+#define LOAD "load"
+#define WORK "work"
+#define DONE "done"
+
+/obj/machinery/neotheology/cruciformforge
+	name = "Cruciform Forge"
+	desc = "This unique piece of technology can be fed biomatter, plasteel, and gold in order to print more cruciforms. Only a select few NeoTheologian architects know how to create these machines. "
+	icon = 'icons/obj/neotheology_machinery.dmi'
```
Comment (Evankhell561) [inline code/modules/core_implant/cruciform/machinery/cruciformforge.dm:8]: there was no specific folder for all machines?

**Review by Evankhell561 — CHANGES_REQUESTED** [review]

File: code/modules/core_implant/cruciform/machinery/cruciformforge.dm (hunk 3 above)
Comment (DarkLevelSS13) [reply code/modules/core_implant/cruciform/machinery/cruciformforge.dm:8]: We have special folder for NT all machines.

File: code/modules/core_implant/cruciform/machinery/cruciformforge.dm (hunk 2 above)
Comment (DarkLevelSS13) [reply code/modules/core_implant/cruciform/machinery/cruciformforge.dm:20]: Forget to remove paused var. My bad.

File: code/modules/core_implant/cruciform/machinery/cruciformforge.dm (hunk 3 above)
Comment (DarkLevelSS13) [reply code/modules/core_implant/cruciform/machinery/cruciformforge.dm:8]: "Cruciform Forge" sprites need to be in neotheology_machinery.dmi, so they can be founded easily.

Comment (Evankhell561) [reply code/modules/core_implant/cruciform/machinery/cruciformforge.dm:8]: I already hit the resolve conversation button.

File: code/modules/core_implant/cruciform/machinery/cruciformforge.dm [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+/obj/machinery/neotheology/cruciformforge/Initialize()
+	. = ..()
+
+	image_load = new(src)
+
+	for(var/_material in needed_material)
+		stored_material[_material] = rand(1, 10)
+
+/obj/machinery/neotheology/cruciformforge/examine(user)
+	..()
```
Comment (Evankhell561) [inline code/modules/core_implant/cruciform/machinery/cruciformforge.dm:37]:
```suggestion
	. = ..()
```

File: code/modules/core_implant/cruciform/machinery/cruciformforge.dm [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+	for(var/_material in stored_material)
+		matter_count += " [stored_material[_material]] [_material]"
+
+	to_chat(user, SPAN_NOTICE("Need materials: [english_list(matter_count_need)].\nIt contains: [english_list(matter_count)]."))
+
+/obj/machinery/neotheology/cruciformforge/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/stack))
+		eat(user, I)
+		return
+
```
Comment (Evankhell561) [inline code/modules/core_implant/cruciform/machinery/cruciformforge.dm:53]:
```suggestion
	. = ..()

```

File: code/modules/core_implant/cruciform/machinery/cruciformforge.dm [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+	use_power(power_cost)
+	working = TRUE
+	start_working = world.time
+	flick_anim(WORK)
+	START_PROCESSING(SSobj, src)
+
+/obj/machinery/neotheology/cruciformforge/Process()
+	if(!working)
+		STOP_PROCESSING(SSobj, src)
```
Comment (Evankhell561) [inline code/modules/core_implant/cruciform/machinery/cruciformforge.dm:66]: SSmachines

File: code/modules/core_implant/cruciform/machinery/cruciformforge.dm [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+/obj/machinery/neotheology/cruciformforge/proc/eat(mob/living/user, obj/item/eating)
+
+	if(!eating && istype(user))
+		eating = user.get_active_hand()
+
+	if(!istype(eating))
+		return FALSE
+
+	if(stat)
```
Comment (Evankhell561) [inline code/modules/core_implant/cruciform/machinery/cruciformforge.dm:83]: comment: I don't understand why you need this (I don't ask you to change it, I just want to know).

File: code/modules/core_implant/cruciform/machinery/cruciformforge.dm [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+/obj/machinery/neotheology/cruciformforge
+	name = "Cruciform Forge"
+	desc = "This unique piece of technology can be fed biomatter, plasteel, and gold in order to print more cruciforms. Only a select few NeoTheologian architects know how to create these machines. "
+	icon = 'icons/obj/neotheology_machinery.dmi'
+	icon_state = "cruciforge"
+
+	density = TRUE
+	anchored = TRUE
+	layer = 2.8
```
Comment (Evankhell561) [inline code/modules/core_implant/cruciform/machinery/cruciformforge.dm:13]: why do you use this layer?

File: code/modules/core_implant/cruciform/machinery/cruciformforge.dm [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+	// A vis_contents hack for materials loading animation.
+	var/tmp/obj/effect/flicker_overlay/image_load
+
+/obj/machinery/neotheology/cruciformforge/Initialize()
+	. = ..()
+
+	image_load = new(src)
+
+	for(var/_material in needed_material)
+		stored_material[_material] = rand(1, 10)
```
Comment (Evankhell561) [inline code/modules/core_implant/cruciform/machinery/cruciformforge.dm:34]: Why not just put the necessary materials on a table next to the machine?

**Review by Evankhell561 — CHANGES_REQUESTED** [review]

File: code/modules/core_implant/cruciform/machinery/cruciformforge.dm (hunk 7 above)
Comment (DarkLevelSS13) [reply code/modules/core_implant/cruciform/machinery/cruciformforge.dm:84]: If machine is broken it usually have stat == 0

Comment (Evankhell561) [reply code/modules/core_implant/cruciform/machinery/cruciformforge.dm:84]: So, according to your code when, will the machine only work when it is broken?

File: code/modules/core_implant/cruciform/machinery/cruciformforge.dm (hunk 9 above)
Comment (DarkLevelSS13) [reply code/modules/core_implant/cruciform/machinery/cruciformforge.dm:34]: There is a tiny space for that. Also, it gives the same feeling as it was used before.

File: code/modules/core_implant/cruciform/machinery/cruciformforge.dm (hunk 8 above)
Comment (DarkLevelSS13) [reply code/modules/core_implant/cruciform/machinery/cruciformforge.dm:13]: This layer commonly used in all machines. Parent of cruciformforge doesn't have this layer.

File: code/modules/core_implant/cruciform/machinery/cruciformforge.dm (hunk 6 above)
Comment (DarkLevelSS13) [reply code/modules/core_implant/cruciform/machinery/cruciformforge.dm:67]: Sorry, forget to change that.

Comment (DarkLevelSS13) [reply code/modules/core_implant/cruciform/machinery/cruciformforge.dm]: Done.

File: code/modules/core_implant/cruciform/machinery/cruciformforge.dm (hunk 7 above)
Comment (DarkLevelSS13) [reply code/modules/core_implant/cruciform/machinery/cruciformforge.dm:84]: Sorry, i misspelling. If it has stat != 0 then it's broken or all what in defines been sad:
// Bitflags for machine stat variable.
#define BROKEN   0x1
#define NOPOWER  0x2
#define POWEROFF 0x4  // TBD.
#define MAINT    0x8  // Under maintenance.
#define EMPED    0x10 // Temporary broken by EMP pulse.

**Review by TheShown911 — APPROVED** [review]

**Review by MadmanMartian — APPROVED** [review]
looks good

**Pull request merged**